src/http/httpClient/KiotaClientFactory.cs

```diff
@@ -26,20 +28,7 @@ public static class KiotaClientFactory
         /// <param name="optionsForHandlers">A array of <see cref="IRequestOption"/> objects passed to the default handlers.</param>
         /// <returns>The <see cref="HttpClient"/> with the default middlewares.</returns>
         public static HttpClient Create(HttpMessageHandler? finalHandler = null, IRequestOption[]? optionsForHandlers = null)
-        {
-            var defaultHandlersEnumerable = CreateDefaultHandlers(optionsForHandlers);
-            int count = 0;
-            foreach(var _ in defaultHandlersEnumerable) count++;
-
-            var defaultHandlersArray = new DelegatingHandler[count];
-            int index = 0;
-            foreach(var handler2 in defaultHandlersEnumerable)
-            {
-                defaultHandlersArray[index++] = handler2;
-            }
```

Comment:
This does a conversion from IList to an array. I replaced it with a collection initializer.
The same applies to the other Create() overload.